README.md

```diff
@@ -20,27 +20,51 @@ automatically installed by the plugin.
 
 # How to use
 
-In your Gradle App Engine Java app, add the following plugin to your build.gradle:
+The plugin needs to be defined in your `build.gradle`. It is directly available on Maven Central. Alternatively, you can download it from GitHub and deploy it to your local repository. The following code snippet shows an example on how to retrieve it from Maven Central:
 
+### Using plugins block
+Since the `appengine-gradle-plugin` is not published to the gradle plugin portal, you must refrence it from the Central Maven repository.  Add the maven central resolution to your `settings.gradle`
 ```Groovy
-apply plugin: 'com.google.cloud.tools.appengine'
+pluginManagement {
+  repositories {
+    gradlePluginPortal()
+    mavenCentral()
+    // add mavenLocal() if you are using a locally built version of the plugin
+  }
+  resolutionStrategy {
+    eachPlugin {
+      if (requested.id.id.startsWith('com.google.cloud.tools.appengine')) {
```

Comment:
Maybe `requested.id.id` --> `requested.id`?

And `resolutionStrategy` and `useModule` are alien to me. Why do we need them? Are they general required whenever someone wants to apply a plugin from the Maven Central?

Reply:
Right, so `requested.id.id` is correct.

Basically there's no mapping from the plugin "com.google.cloud.tools.appengine" to the artifact. So here we create that mapping.

[`ResolutionStrategy.eachPlugin`](https://github.com/gradle/gradle/blob/674c1da6b12978247560a3223670cc1835482f1c/subprojects/plugin-use/src/main/java/org/gradle/plugin/management/internal/DefaultPluginResolutionStrategy.java#L46)

uses the type [`PluginResolveDetails`](https://github.com/gradle/gradle/blob/1293df631c1a20647a0489175d7aa63d60d65e30/subprojects/core-api/src/main/java/org/gradle/plugin/management/PluginResolveDetails.java#L32)

where we get a [`PluginRequest`] (https://github.com/gradle/gradle/blob/1293df631c1a20647a0489175d7aa63d60d65e30/subprojects/core-api/src/main/java/org/gradle/plugin/management/PluginRequest.java#L36)

And the first `id` is the `id` of the request
in the style
```
plugins {
  id `the-id-in-question` version `...`
}
```
This `id` is actually a [`PluginId`](https://github.com/gradle/gradle/blob/1293df631c1a20647a0489175d7aa63d60d65e30/subprojects/core-api/src/main/java/org/gradle/plugin/use/PluginId.java)

So we obtain the `id` from this `id`, and hence the nested `requested.id.id`

Reply:
> Are they general required whenever someone wants to apply a plugin from the Maven Central?

Also to answer this no. We can publish a plugin to maven central in the same way as one would to gradle plugin repo, but it's a two step process.

1. Publish the actual jar with code: https://plugins.gradle.org/m2/gradle/plugin/com/google/cloud/tools/jib-gradle-plugin/2.5.0/
2. Publish the "plugin jar" following some gradle convention with a link to the jar in (1): 
https://plugins.gradle.org/m2/com/google/cloud/tools/jib/com.google.cloud.tools.jib.gradle.plugin/2.5.0/

So I think gradle can do the same for other repos like maven central -- as long as we follow the convention?

Reply:
Now I get it. The Gradle plugin model requiring a separate "plugin jar" seems a bit weird to me, but I guess there must be reason.

I'm curious why the App Engine plugin is not published to the Gradle Plugin Portal but to Maven Central (like Jib)?